ml-agents/mlagents/trainers/env_manager.py

```diff
@@ -101,9 +114,13 @@ def advance(self):
                     _policy = self.agent_managers[brain_name].policy_queue.get_nowait()
             except AgentManagerQueue.Empty:
                 if _policy is not None:
-                    self.set_policy(brain_name, _policy)
-        # Step the environment
+                    # policy_queue contains Policy, but we need a TFPolicy here
+                    self.set_policy(brain_name, _policy)  # type: ignore
```

Comment:
We weren't previously typchecking this method. This was an old "error" but is surfaced now. Not sure if we should fix it here or separately. cc @ervteng @andrewcoh

Reply:
I think there was some subtlety about types here given that Policy didn't contain some of the methods used by the AgentProcessor, e.g. get/set memories. We can fix this after Arthur's PR #4254 is merged.

Reply:
Sounds good, I'll leave the `# type: ignore` for now.